message_sender/tasks.py

```diff
@@ -77,6 +79,44 @@ def run(self, metric_name, metric_value, session=None, **kwargs):
 fire_metric = FireMetric()
 
 
+class Concurrency_Limiter(object):
+    def __init__(self):
+        self.redis_server = redis.StrictRedis(host=settings.REDIS_HOST,
+                                              port=settings.REDIS_PORT,
+                                              db=settings.REDIS_DB)
+
+    def get_current_message_count(self, msg_type):
+        keys = self.redis_server.keys(pattern=msg_type + "_messages_at_*")
+        total = 0
+        # Sum the values in all the buckets to get the total
+        for key in keys:
+            total += int(self.redis_server.get(key))
+        return total
+
+    def incr_message_count(self, msg_type, delay):
+        # Buckets of 1min
+        key = msg_type + "_messages_at_" + time.strftime("%M")
+        value = self.redis_server.incr(key, 1)
+        if value == 1:
+            # Add 60 seconds to the expiry time so messages that start at the
+            # end of the minute still complete
+            self.redis_server.expire(key, delay + 60)
```

Comment:
We might want to have `BUCKET_SIZE = 60` as a class variable, or a settings variable, in case we need to change it, and to avoid magic numbers.